scripts/migration/v0.0.4__add-room-templates.ts

```diff
@@ -0,0 +1,119 @@
+import { MigrateOptions } from "fireway";
+import { isEqual } from "lodash";
+
+import { getVenueId } from "../../functions/src/utils/venue";
+import { Room } from "../../src/types/rooms";
+
+const isVenueRegex = /\/in\/(\w+)$/;
+
+interface CreateVenueFromPortalProps {
+  firestore: FirebaseFirestore.Firestore;
+  portal: Room;
+  parentVenueId: string;
+  parentWorldId: string;
+}
+
+const createVenueFromPortal = async ({
+  firestore,
+  portal,
+  parentVenueId,
+  parentWorldId,
+}: CreateVenueFromPortalProps) => {
+  const venueData = {
+    name: portal.title,
+    template: "zoomroom",
+    zoomUrl: portal.url ?? "",
+    parentId: parentVenueId,
+    worldId: parentWorldId,
+  };
+
+  let venueExists = true;
+  let index = -1;
+  const baseVenueId = getVenueId(portal.title);
+  let venueId = baseVenueId;
+  while (venueExists) {
+    index += 1;
+    venueId =
+      (baseVenueId || "externalexperience") + (index === 0 ? "" : index);
+    const venue = await firestore.collection("venues").doc(venueId).get();
+    venueExists = venue.exists;
+  }
+
+  await firestore.collection("venues").doc(venueId).set(venueData);
+
+  return `/in/${venueId}`;
+};
+
+export const migrate = async ({ firestore }: MigrateOptions) => {
+  const { docs: venueDocs } = await firestore
+    .collection("venues")
+    .where("template", "==", "partymap")
+    .get();
+
+  await Promise.all(
+    venueDocs.map(async (venueDoc) => {
+      const rooms: Room[] = venueDoc.data().rooms ?? [];
+      await Promise.all(
+        rooms?.map(async (room) => {
+          if (room.template === "audience") {
```

Comment:
I think extracting the inside of this `map` into a function would make this code clearer. It's quite long and there's a lot going on. e.g.
```
await Promise.all(rooms?.map(transformRoomAndCreateVenues))
```

Reply:
done here: https://github.com/sparkletown/sparkle/pull/2500/commits/3b7e19c6c5d571a8eed048f0d68a5f5771e56a7b